server/src/main/java/com/google/coffeehouse/servlets/GetClubServlet.java

```diff
@@ -0,0 +1,107 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.servlets;
+
+import com.google.coffeehouse.common.Club;
+import com.google.coffeehouse.storagehandler.StorageHandlerApi;
+import com.google.coffeehouse.storagehandler.StorageHandler;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet to get a {@link Club} from Http GET Request Body (in JSON format)
+ * that exists in the database through a call to the Storage Handler API,
+ * and return it in JSON format.
+ */
+@WebServlet("/api/get-club")
+public class GetClubServlet extends HttpServlet {
+  /** 
+   * The error string sent by the response object in doGet when the body of the
+   * GET request cannot be used to fetch a {@link Club} for any reason.
+   */
+  public static final String BODY_ERROR = "- unable to parse body.";
+  /** The logged error string when an error parsing the body of the GET request is encountered. */
+  public static final String LOG_BODY_ERROR_MESSAGE =
+      "Body unable to be parsed in GetClubServlet: ";
+  /** Message to be logged when the body of the GET request does not have required fields. */
+  public static final String LOG_INPUT_ERROR_MESSAGE =
+      "Error with JSON input in GetClubServlet: No \"%s\" found in JSON.";
+
+  private static final Gson gson = new Gson();
+  private final StorageHandlerApi storageHandler;
+
+  /**
+   * Overloaded constructor for dependency injection.
+   * @param storageHandler the {@link StorageHandlerApi} that is used when fetching the Club
+   */
+  public GetClubServlet(StorageHandlerApi storageHandler) {
+    super();
+    this.storageHandler = storageHandler;
+  }
+
+  /**
+   * Explicit default constructor used for instantiating the servlet when not testing.
+   */
+  public GetClubServlet() {
+    super();
+    this.storageHandler = new StorageHandlerApi();
+  }
+  
+  /** 
+   * Returns a {@link Club} object in JSON format from information in the database.
+   * @param request the GET request that must have a valid JSON representation of the "clubId" to
+   *     be passed in order to fetch a Club from ID in the database. If this is not the case the
+   *     response will send a "400 Bad Request error". If an error occurs when attempting to fetch
+   *     the Club from the database, the response will send a "500 Internal Server error" 
+   * @param response the response from this method, will contain the object in JSON format.
+   *     If the request object does not have a valid JSON body that describes the Club, this object
+   *     will send a "400 Bad Request error" If an error occurs when attempting to fetch
+   *     the Club from the database, this object will send a "500 Internal Server error" 
```

Comment:
Make sure to add the SC_NOT_FOUND error which is encountered if the club does not exist in the database. Something along the lines of: "If the request object is attempting to get a club which does not exist in the database, the response will send a "404 Not Found error".

Reply:
Good catch! This has been added and pushed.